clvm/serialize.py

```diff
@@ -124,6 +124,8 @@ def _atom_from_stream(f, b, to_sexp):
             raise ValueError("bad encoding")
         size_blob += b
     size = int.from_bytes(size_blob, "big")
+    if size >= 0x400000000:
```

Comment:
I'm pretty sure this was a DoS vector. however, it still might be. I think we should consider lowering the max bob size

Reply:
Yeah, it's pretty big.

Reply:
For sure if it does a malloc, this is a DOS and the stream could crash. We ought to test this. I think in practice, the read is done on a `io.BytesIO` object, which might grab a substring of an existing string, so we'd just see an end-of-stream error (since the stream can't be long enough).

Reply:
If we can verify this as a DOS vector, we might consider adding an extra parameter `max_atom_length` or something to the deserialize code. If we make it larger than the maximum generator size, the exact value is not consensus-critical.

Reply:
(I've also considered writing the deserializer in clvm, which is possible now that we have `substr` and `len` and the bitfield operators. Then, our atoms would just be pointers into the generator bock as long substrings are shared by `substr`.)

Reply:
Hmmm, this number is ~17 GB, so I'm not this change actually helps on machines with less than 17 GB of free memory (which is most computers).

Reply:
yeah, this is the most conservative limit. Since we have this limit in the encoder, we would reasonably not support parsing atoms this large. This is not really meant to be an attempt to address the potential DoS

Reply:
I realize now that this limit already exists in serializing, so adding it to deserializing is about making things symmetric, not DOS mitigation.